pyas2/tests/test_basic.py

```diff
@@ -534,6 +534,68 @@ def testEncryptSignMessageAsyncSignMdn(self, mock_request):
         mock_request.side_effect = RequestException()
         out_message.mdn.send_async_mdn()
 
+    @mock.patch("requests.post")
+    def testForceBinaryCanonicalization(self, mock_request):
+        """Test Permutation 15: Sender sends text data using binary canonicalization
+        and requests a synchronous receipt."""
+
+        partner = Partner.objects.create(
+            name="AS2 Server",
+            as2_name="as2server",
+            target_url="http://localhost:8080/pyas2/as2receive",
+            signature="sha1",
+            signature_cert=self.server_crt,
+            encryption=None,
+            encryption_cert=None,
+            mdn=True,
+            mdn_mode="SYNC",
+        )
+
+        receiver = Partner.objects.get(as2_name="as2client")
+        receiver.canonicalize_as_binary = True
+        receiver.save()
+
+        as2message = As2Message(
+            sender=self.organization.as2org, receiver=partner.as2partner
+        )
+
+        with mock.patch("pyas2lib.as2.canonicalize") as mock_canonicalize:
+            mock_canonicalize.side_effect = self.mock_canonicalize_function
+
+            as2message.build(
+                self.payload,
```

Comment:
we are using the same payload as the other tests so how is canonicalize_as_binary affecting this?

Reply:
i would choose the first option here because the intent of the test is more clear

Reply:
Sure, I can change it - not sure I can make it before May though.